portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,13 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+function fetchMessage() {
+    fetch("/data").then(request => request.json()).then((textResponse) => {
+        for (var i = 0; i < 3; i++) {
+            var paragraph_tag = document.createElement("p");
+            paragraph_tag.innerText = textResponse[i];
+            document.getElementById("message-container").appendChild(paragraph_tag);
```

Comment:
Let's move the getElementById call to outside the for-loop to avoid looking for the element every time the loop is executed.

Reply:
I have moved the variable out of the loop.